packages/edit-site/src/components/save-button/index.js

```diff
@@ -78,7 +81,27 @@ export default function SaveButton( {
 		}
 		return __( 'Save' );
 	};
-	const label = getLabel();
+
+	/**
+	 * We focus on adding the customization to the SaveButton's `onClick` and `label` for now.
+	 * We will provide the customization to the other entry points (e.g., SavePanel, SaveHub) in the future if needed.
+	 * @see https://github.com/WordPress/gutenberg/pull/56807
+	 */
+	const onClick = useCallback( () => {
+		const callback = applyFilters( 'edit-site.SaveButton.onClick', () =>
+			setIsSaveViewOpened( true )
+		);
+		callback();
+	}, [ setIsSaveViewOpened ] );
+	const label = useMemo( () => {
+		return applyFilters( 'edit-site.SaveButton.label', getLabel(), {
+			isSaving,
+			disabled,
+			isPreviewingTheme,
```

Comment:
It might be better to add it to the deps as well even if it won't be changed.

But the naming is still confusing as people may not notice it's a function instead of the boolean value. However, it's out of the scope of this PR 🙂